src/dmd/dmodule.d

```diff
@@ -529,6 +539,38 @@ extern (C++) final class Module : Package
             fprintf(global.stdmsg, "%s\t(%s)\n", ident.toChars(), m.srcfile.toChars());
         }
         m = m.parse();
+
+        // Call onImport here because if the module is going to be compiled then we
+        // need to determine it early because it affects semantic analysis. This is
+        // being done after parsing the module so the full module name can be taken
+        // from whatever was declared in the file.
+
+        /*
```

Comment:
Anyone have any ideas on why this segfaults on osx 32bit?  The code below doesn't segfault but I'm not sure why.